src/DonorDashboards/Tabs/EditProfileTab/AvatarRoute.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Give\DonorDashboards\Tabs\EditProfileTab;
+
+use WP_REST_Request;
+use Give\DonorDashboards\Tabs\Contracts\Route as RouteAbstract;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class AvatarRoute extends RouteAbstract {
+
+	/**
+	 * @inheritdoc
+	 */
+	public function endpoint() {
+		return 'avatar';
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function args() {
+		return [];
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @return array
+	 */
+	public function handleRequest( WP_REST_Request $request ) {
+
+		if ( ! ( is_array( $_POST ) && is_array( $_FILES ) ) ) {
+			return [];
+		}
+
+		// Delete existing Donor profile avatar attachment
+		if ( give()->donorDashboard->getAvatarId() ) {
+			wp_delete_attachment( give()->donorDashboard->getAvatarId(), true );
+		}
+
+		if ( ! function_exists( 'wp_handle_upload' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/file.php' );
+		}
+
+		foreach ( $_FILES as $file ) {
+			$upload = wp_handle_upload(
+				$file,
+				[
+					'test_form' => false,
+				]
+			);
+
+			if ( isset( $upload['url'] ) ) {
+				$path = $upload['url'];
+
+				// Check the type of file. We'll use this as the 'post_mime_type'.
+				$filetype = wp_check_filetype( basename( $path ), null );
+
+				// Prepare an array of post data for the attachment.
+				$attachment = [
+					'guid'           => $path,
+					'post_mime_type' => $filetype['type'],
+					'post_title'     => preg_replace( '/\.[^.]+$/', '', basename( $path ) ),
+					'post_content'   => '',
+					'post_status'    => 'inherit',
+				];
+
+				// Insert the attachment.
+				$attachmentId = wp_insert_attachment( $attachment, $path );
+
+				// Make sure that this file is included, as wp_generate_attachment_metadata() depends on it.
+				require_once( ABSPATH . 'wp-admin/includes/image.php' );
+
+				// Generate the metadata for the attachment, and update the database record.
+				$attachmentData = wp_generate_attachment_metadata( $attachmentId, $path );
+				wp_update_attachment_metadata( $attachmentId, $attachmentData );
+
+				return [
+					'id' => $attachmentId,
+				];
+			}
+		}
+
+		return [];
```

Comment:
Is it possible to reach this?

Reply:
Ah, good point! Yeah, gone ahead and removed this return statement: https://github.com/impress-org/givewp/pull/5745/commits/b3f2cca77a6dc5c26dbc4eb8c9eaa16f06bfc4ab